src/components/Registration.jsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Card
+} from "@mui/material";
+import React, { useState, useEffect } from "react";
+import { StyledCardContent, StyledButton, StyledForm, StyledTitle, Heading, StyledTextField } from '../styles/styled';
+
+export const Registration = ({ onRegistration }) => {
+  const [user, setUser] = useState({
+    username: "",
+    email: "",
+    password: "",
+    passwordConfirm: "",
+  });
+
+  const [isValidForm, setIsValidForm] = useState(false);
+
+  useEffect(() => {
+    if (validateEmail(user.email) && user.password && onConfirmPassword()) {
+      setIsValidForm(true);
+    } else {
+      setIsValidForm(false);
+    }
+  }, [user]);
+
+  const validateEmail = (email) => {
+    return /^\S+@\S+\.\S+$/.test(email);
+  };
+
+  const extractUsername = (email) => {
+    return email.slice(0, email.indexOf("@"));
+  };
+
+  const onChangeEmail = (e) => {
+    if (validateEmail(e.target.value)) {
+      setUser({
+        ...user,
+        username: extractUsername(e.target.value),
+        email: e.target.value,
+      });
+    } else {
+      setUser({
+        ...user,
+        username: "",
+        email: e.target.value,
+      });
+    }
+  };
+
+  const onConfirmPassword = () => {
+    return user.password === user.passwordConfirm;
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    console.log(user);
+    onRegistration();
```

Comment:
to console.log by mělo být součástí té onRegistration funce :) (ale už jen logická drobnost)